### ISSUE TITLE
`iter_start_seqnum` and `preserve_deletes` are going to be deprecated (use user-defined timestamp instead)

### ISSUE THREAD

**Issue by jay-zhuang** [facebook/rocksdb#9090]
## tldr
[`ReadOptions::iter_start_seqnum`](https://github.com/facebook/rocksdb/blob/6d93b87588ecf34ad281dc881cdf27edcde09c9b/include/rocksdb/options.h#L1552) and [`DBOptions::preserve_deletes`](https://github.com/facebook/rocksdb/blob/6d93b87588ecf34ad281dc881cdf27edcde09c9b/include/rocksdb/options.h#L1209) are going to be deprecated, please try using [user defined timestamp feature](https://github.com/facebook/rocksdb/wiki/User-Timestamp-%28Experimental%29) instead.

## details
[`ReadOptions::iter_start_seqnum`](https://github.com/facebook/rocksdb/blob/6d93b87588ecf34ad281dc881cdf27edcde09c9b/include/rocksdb/options.h#L1552) and [`DBOptions::preserve_deletes`](https://github.com/facebook/rocksdb/blob/6d93b87588ecf34ad281dc881cdf27edcde09c9b/include/rocksdb/options.h#L1209) are used to support differential snapshots. Currently the feature is not well maintained (some APIs are not fully supported, e.g. #6837, #8472), and the interface is not user friendly, which returns rocksdb internal key from the iterator. There's a new [user defined timestamp feature](https://github.com/facebook/rocksdb/wiki/User-Timestamp-%28Experimental%29) introduced, which is to support similar feature. User defined timestamp is more flexible and has better user interface. Please switching to that if you're using these feature. The deprecated feature will be removed in a future release.
